mmv1/third_party/terraform/services/compute/resource_compute_attached_disk.go.erb

```diff
@@ -38,7 +38,6 @@ func ResourceComputeAttachedDisk() *schema.Resource {
 
 		CustomizeDiff: customdiff.All(
 			tpgresource.DefaultProviderProject,
-			tpgresource.DefaultProviderZone,
```

Comment:
Rather than remove this outright, we likely want a resource-specific version that handles pulling the value out of the instance id. @NickElliot in case you have any thoughts.

Reply:
Yeah, it seems like a pretty straight forward change I could go ahead and takeover if you want @edwardmedia

Reply:
@rileykarson @NickElliot the issue might be at here https://github.com/hashicorp/terraform-provider-google/blob/a3205a9f9416ac58a0c8e9898a063c7e3181923f/google/tpgresource/utils.go#L793. The value of `zone` is actually "", not nil in this case during the plan.

Reply:
I think that line is just checking the resource has a field called zone since ResourceData/ResourceDiff methods will only return nil if a field does not appear in a resource schema! @NickElliot can you confirm? (also, we may want to comment those lines if that's the case since it's nonobvious)

Reply:
correct, if it doesnt exist in schema it returns nil, if it does exist in schema it will return the type-equivalent of nothing aka "" for strings, 0 for ints. 

I'll add in the comments with my fix for this